indexer-app/test/cmr/indexer/test/data/index_set_generics.clj

```diff
@@ -0,0 +1,48 @@
+(ns cmr.indexer.test.data.index-set-generics
+  (:require
+   [clojure.test :refer :all]
+   [cmr.indexer.data.index-set-generics :as gen]
+   [cmr.system-int-test.utils.dev-system-util :as dev-sys-util]))
+
+(def index-definition-def 
+  {:MetadataSpecification 
+   {:URL "https://cdn.earthdata.nasa.gov/generics/dataqualitysummary/v1.0.0"
+    :Name "DataQuailtySummary",
+    :Version "1.0.0"} 
+   :SubConceptType "DQS", 
+   :IndexSetup {:index {:number_of_shards 5, :number_of_replicas 1, :refresh_interval "1s"}},
+   :Indexes [{:Description "Name", :Field ".Name", :Name "Name", :Mapping "string"}
+             {:Description "Id - This is the old ECHO 10 GUID.", :Field ".Id", :Name "Id", :Mapping "string"}]})
+
+(deftest get-settings-test
+  "Test getting the settings from either the config file, the environment variable, or the default value." 
+
+  (testing "Test that concept prefixes can be looked up from either a configuration file or be assumed"
+    (is (= {:index
+            {:number_of_shards 5, :number_of_replicas 1, :refresh_interval "1s"}}
+           (gen/get-settings index-definition-def))))
+ 
+  (testing "Test that a different shard number in the configuration file takes precidence." 
+    (let [index-def-2 (assoc-in index-definition-def [:IndexSetup :index :number_of_shards] 4)
+          actual (gen/get-settings index-def-2)]
+      (is (= (get-in index-def-2 [:IndexSetup :index :number_of_shards])
+             (get-in actual [:index :number_of_shards]))))) 
+
+  (testing "Test that setting the environment variable takes precidence"
+    (dev-sys-util/eval-in-dev-sys `(gen/set-elastic-generic-index-num-shards! 3))
+    (is (= 3 (get-in (gen/get-settings index-definition-def) [:index :number_of_shards])))
+    ;; Saving the original value doesn't work, because you can't use the variable to set it back. So the default is being used.
+    (dev-sys-util/eval-in-dev-sys `(gen/set-elastic-generic-index-num-shards! gen/default-generic-index-num-shards))))
+
+(deftest read-schema-definition-test
+  "Testing the read schema functionality"
```

Comment:
Same as above, regarding `deftest` docstrings

Reply:
I have put the text as a comment.